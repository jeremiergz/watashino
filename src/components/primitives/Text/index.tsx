import styled, { StyledComponentProps } from 'styled-components';
import {
    color,
    ColorProps,
    compose,
    display,
    DisplayProps,
    space,
    SpaceProps,
    typography,
    TypographyProps,
    zIndex,
    ZIndexProps,
} from 'styled-system';
import { Theme } from '../../../theme';

type TextProps = StyledComponentProps<
    'span',
    Theme,
    ColorProps & DisplayProps & SpaceProps & Omit<TypographyProps, 'fontWeight'> & ZIndexProps,
    never
> & {
    as?: keyof JSX.IntrinsicElements | React.ComponentType<unknown>;
    color?: keyof Theme['colors'];
    fontWeight?: keyof Theme['fontWeights'];
};

const Text = styled.span<TextProps>`
    ${compose(
        color,
        display,
        space,
        typography,
        zIndex,
    )}
`;

Text.displayName = 'Text';

export { TextProps };
export default Text;
