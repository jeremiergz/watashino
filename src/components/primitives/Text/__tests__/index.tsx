import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Text from '..';

describe('components/primitives/Text Test Suite', () => {
  it('matches snapshot', () => {
    const jsx = <Text>Text Component</Text>;
    expect(ReactTestRenderer.create(jsx).toJSON()).toMatchSnapshot();
  });
});
