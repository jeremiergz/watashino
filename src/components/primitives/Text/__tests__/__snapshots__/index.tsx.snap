// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/primitives/Text Test Suite matches snapshot 1`] = `
<span
  className="sc-bdVaJa ibTDPh"
>
  Text Component
</span>
`;
